src/validation.cpp

```diff
@@ -4348,12 +4349,29 @@ void CChainState::CheckBlockIndex()
     while (pindex != nullptr) {
         nNodes++;
         if (pindexFirstInvalid == nullptr && pindex->nStatus & BLOCK_FAILED_VALID) pindexFirstInvalid = pindex;
-        if (pindexFirstMissing == nullptr && !(pindex->nStatus & BLOCK_HAVE_DATA)) pindexFirstMissing = pindex;
+        if (pindexFirstMissing == nullptr && !(pindex->nStatus & BLOCK_HAVE_DATA) && !pindex->IsAssumedValid()) {
```

Comment:
In commit "validation: set BLOCK_ASSUMED_VALID during snapshot load" (9a8ad497238c8c8aba1ba8679bc7cdbe409c635e)

Would be nice to have comment explaining the IsAssumedValid condition, since logic is a little convoluted. Maybe "Treat assumed valid blocks that are missing data as if they have data for the purpose of the [whatever] check below"